### PR TITLE
Add automated release workflow

### PR DESCRIPTION
**Pull request by frederikprijck**
### Changes

This adds the automated release workflow based on what we have in other .NET repositories.
The main difference here is that this repository is a mono-repository, publishing multiple packages.

Therefore, our nuget-release workflow and nuget-publish action are different from the other .NET repositories.

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [x] All existing and new tests complete without errors
- [x] All code guidelines in the [CONTRIBUTING documentation](../CONTRIBUTING.md) have been run/followed

### CODE REVIEW

File: .github/actions/nuget-publish/action.yml [hunk 1]
```diff
@@ -9,23 +9,32 @@ inputs:
...
+      uses: nuget/setup-nuget@v1
     
     - name: Create NuGet packages
       shell: pwsh
-      run: dotnet pack "${{ inputs.project-path }}" --configuration Release --output ${{ inputs.nuget-directory }}
+      run: nuget pack ${{ inputs.nuspec-file }} -OutputDirectory ${{ inputs.nuget-directory }}
+
+    - name: Create NuGet packages
+      shell: bash
+      run: ls ${{ inputs.nuget-directory }}
```
Comment (frederikprijck) [inline .github/actions/nuget-publish/action.yml:32]:
```suggestion
```

File: .github/workflows/nuget-release.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - uses: actions/download-artifact@v4
+      with:
+        path: './src'
+        name: build
+
+    - id: release_ios
```
Comment (frederikprijck) [inline .github/workflows/nuget-release.yml:41]:
```suggestion
    - id: release
```

File: .github/actions/nuget-release/action.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        nuget-directory: ${{ inputs.nuget-directory }}
+        nuspec-file: ${{ inputs.nuspec-file }}
+
+    # Create a release for the tag
+    - uses: ./.github/actions/release-create
+      if: steps.tag_exists.outputs.exists == 'false'
+      with:
+        token: ${{ inputs.github-token }}
+        name: ${{ steps.get_version.outputs.version }}
+        # body: ${{ steps.get_release_notes.outputs.release-notes }}
```
Comment (ewanharris) [inline .github/actions/nuget-release/action.yml:70]: Should we remove this and the get-release-notes step for now?

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    environment: release
+    needs: build
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: actions/download-artifact@v4
```
Comment (ewanharris) [inline .github/workflows/release.yml:169]: Is it worth downgrading this to v3 as there seems to be issues amongst the latest *-artifact action releases

```suggestion
      - uses: actions/download-artifact@v3
```

File: .github/workflows/nuget-release.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    runs-on: windows-latest
+    environment: release
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - uses: actions/download-artifact@v4
```
Comment (ewanharris) [inline .github/workflows/nuget-release.yml:36]: Is it worth downgrading this to v3 as there seems to be issues amongst the latest *-artifact action releases
```suggestion
    - uses: actions/download-artifact@v3
```

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+      - name: Restore NuGet
+        run: nuget restore Auth0.OidcClient.All.sln
+
+      - name: Setup MSBuild
+        uses: microsoft/setup-msbuild@v1.1
+
+      - name: Build
+        run: msbuild Auth0.OidcClient.All.sln -t:rebuild -verbosity:diag -property:Configuration=Release
+
+      - uses: actions/upload-artifact@v4
```
Comment (ewanharris) [inline .github/workflows/release.yml:53]: Is it worth downgrading this to v3 as there seems to be issues amongst the latest *-artifact action releases
```suggestion
      - uses: actions/upload-artifact@v3
```

**Review by ewanharris — CHANGES_REQUESTED** [review]

File: .github/actions/nuget-release/action.yml (hunk 3 above)
Comment (frederikprijck) [reply .github/actions/nuget-release/action.yml:70]: Oh no, we should keept it !

File: .github/workflows/release.yml (hunk 4 above)
Comment (ewanharris) [reply .github/workflows/release.yml:169]: Wont do this

**Review by ewanharris — APPROVED** [review]

**Pull request merged**